ui/components/DependenciesView.tsx

```diff
@@ -0,0 +1,141 @@
+          There are no dependencies set up for your kustomizations or helm
+          releases at this time. You can set them up using the dependsOn field
+          on the kustomization or helm release object.
+        </div>
+
+        <h2>What are dependencies for?</h2>
+
+        <div>
+          Dependencies allow you to relate different kustomizations and helm
+          releases, as well as specifying an order in which your resources
+          should be started. For example, you can wait for a database to report
+          as 'Ready' before attempting to deploy other services.
+        </div>
+      </Flex>
+    </Flex>
+  );
+}
+
+const Message = styled(UnstyledMessage)`
+  & {
+    ${Flex} {
+      box-sizing: border-box;
+      width: 560px;
+      padding-top: ${(props) => props.theme.spacing.medium};
+      padding-right: ${(props) => props.theme.spacing.xl};
+      padding-bottom: ${(props) => props.theme.spacing.xxl};
+      padding-left: ${(props) => props.theme.spacing.xl};
+      margin-top: ${(props) => props.theme.spacing.xxl};
+      margin-bottom: ${(props) => props.theme.spacing.xxl};
+      border-radius: 10px;
+      background-color: #ffffffd9;
+      color: ${(props) => props.theme.colors.neutral30};
+
+      h2 {
+        margin-top: 0;
+        margin-bottom: 0;
+        font-size: ${(props) => props.theme.fontSizes.large};
+        font-weight: 600;
+
+        &:not(:first-child) {
+          margin-top: ${(props) => props.theme.spacing.medium};
+        }
+      }
+
+      div {
+        margin-top: ${(props) => props.theme.spacing.medium};
+        font-size: 16px;
+        font-weight: 400;
+        line-height: 20px;
+      }
+    }
+  }
+`;
+
+type DependenciesViewProps = {
+  className?: string;
+  automation?: Automation;
+};
+
+const graphNodesPlaceholder = [] as FluxObjectNode[];
+
+function DependenciesView({ className, automation }: DependenciesViewProps) {
+  const [graphNodes, setGraphNodes] = React.useState<FluxObjectNode[] | null>(
+    null
+  );
+
+  const automationKind = automation?.kind;
+
+  const {
+    data,
+    isLoading: isLoadingData,
+    error,
+  } = automation
+    ? useListObjects("", fluxObjectKindToKind(automationKind))
+    : { data: { objects: [], errors: [] }, error: null, isLoading: false };
+
+  React.useEffect(() => {
+    if (isLoadingData) {
+      return;
+    }
+
+    if (error || data?.errors?.length > 0) {
+      setGraphNodes(graphNodesPlaceholder);
+      return;
+    }
+
+    const nodes = getGraphNodes(
+      data.objects.map((obj) => new FluxObjectNode(obj)),
+      automation
+    );
+
+    if (nodes.length === 0) {
+      setGraphNodes(graphNodesPlaceholder);
+    } else {
+      setGraphNodes(nodes);
+    }
+  }, [isLoadingData, data, error]);
+
+  const isLoading = isLoadingData && !graphNodes;
+
+  const shouldShowGraph = !!graphNodes && graphNodes.length > 0;
+
+  return (
+    <RequestStateHandler loading={isLoading} error={error}>
+      {!isLoading && (
```

Comment:
You might not need this loading condition here - I haven't used the RequestStateHandler component before (maybe I should've) - but it seems like it's handling the loading case. If you still need the loading condition then the RequestStateHandler thing doesn't seem like it's doing it's job and should probably be removed

Reply:
I thought RequestStateHandler is used mostly for displaying a loading indicator or an error message. Let me check how it works if I remove the inner condition.

Reply:
Good catch, removed the unneeded inner loading condition, the loading indicator is displayed without it and the content is hidden, while loading data, as expected.

The RequestStateHandler was used in Reconciliation graph, it is where I picked it from.